rudder-core/src/main/scala/com/normation/rudder/services/policies/InterpolatedValueCompiler.scala

```diff
@@ -253,31 +336,56 @@ class InterpolatedValueCompilerImpl extends RegexParsers with InterpolatedValueC
 
   //identifier for step in the path or param names
-  def propId: Parser[String] = """[\-_a-zA-Z0-9]+""".r
+  def propId      : Parser[String] = """[\-_a-zA-Z0-9]+""".r
+
+  // all interpolation
+  def interpol    : Parser[Interpolation] = rudderProp | nodeProp
 
   //an interpolated variable looks like: ${rudder.XXX}, or ${RuDder.xXx}
-  def interpol: Parser[Interpolation] = "${" ~> id("rudder") ~> "." ~> (nodeProp | parameter ) <~ "}"
+  def rudderProp  : Parser[Interpolation] = "${" ~> space ~> id("rudder") ~> space ~> id(".") ~> space ~>(nodeAccess | parameter ) <~ space <~ "}"
 
   //a node path looks like: ${rudder.node.HERE.PATH}
-  def nodeProp: Parser[Interpolation] = {
-    id("node") ~> "." ~> repsep(propId, ".") ^^ { seq => NodeAccessor(seq) }
-  }
+  def nodeAccess  : Parser[Interpolation] = { id("node") ~> space ~> id(".") ~> space ~> repsep(propId, space ~> id(".") ~> space) ^^ { seq => NodeAccessor(seq) } }
 
   //a parameter looks like: ${rudder.PARAM_NAME}
-  def parameter: Parser[Interpolation] = {
-    id("param") ~> "." ~> propId ^^ { p => Param(p) }
-  }
+  def parameter   : Parser[Interpolation] = { id("param") ~> space ~> id(".") ~> space ~> propId ^^ { p => Param(p) } }
+
+  //an interpolated variable looks like: ${rudder.XXX}, or ${RuDder.xXx}
+  def nodeProp    : Parser[Interpolation] = "${" ~> space ~> id("node") ~> space ~> id(".") ~> space ~> id("properties") ~> rep1( space ~> "[" ~> space ~>
+                                            propId <~ space <~ "]" ) ~ opt(propOption) <~ space <~ "}" ^^ { case ~(path, opt) => Property(path, opt) }
+
+  //here, the number of " must be strictly decreasing - ie. triple quote before
+  def propOption  : Parser[PropertyOption] = space ~> "|" ~> space ~>  ( onNodeOpt |  "default" ~> space ~> "=" ~> space ~>
+                                             ( interpolOpt | emptyTQuote | tqString | emptySQuote | sqString )  )
+
+  def onNodeOpt   : Parser[InterpreteOnNode.type] = id("node") ^^^ InterpreteOnNode
+
+  def interpolOpt : Parser[DefaultValue] = interpol ^^ { x => DefaultValue(x::Nil) }
+  def emptySQuote : Parser[DefaultValue] = id("\"\"")         ^^   { _ => DefaultValue(CharSeq("")::Nil) }
+  def emptyTQuote : Parser[DefaultValue] = id("\"\"\"\"\"\"") ^^   { _ => DefaultValue(CharSeq("")::Nil) }
 
+  //string must be simple or triple quoted string
+  def sqString    : Parser[DefaultValue] = id("\"")  ~> rep1(sqplainStr | interpol )<~ id("\"") ^^ { case x => DefaultValue(x) }
+  def sqplainStr  : Parser[Token]        = ("""(?iums)((?!(\Q${\E\s*rudder\s*\.|\Q${\E\s*node\s*\.\s*properties|")).)+""").r  ^^ { str => CharSeq(str) }
 
+  def tqString    : Parser[DefaultValue]   = id("\"\"\"")  ~> rep1(tqplainStr | interpol )<~ id("\"\"\"") ^^ { case x => DefaultValue(x) }
+  def tqplainStr  : Parser[Token]          = ("""(?iums)((?!(\Q${\E\s*rudder\s*\.|\Q${\E\s*node\s*\.\s*properties|"""+"\"\"\""+ """)).)+""").r  ^^ { str => CharSeq(str) }
```

Comment:
Simplified by factoring out the main regex in a common "plainStringRegex". It helps understanding that most of the regex is to avoid "${rudder." and "${node.properties", and that optionnally we are adding " or """ in the avoided items.